Blizzard.IO.RabbitMQ/NetqRabbitConsumer.cs

```
@@ -0,0 +1,93 @@
+            ILoggerFactory loggerFactory, IConverter<MessageProperties, RabbitMessageProperties> coverter = null)
+            : this(netqBus, sourceQueue, loggerFactory, coverter)
+        {
+            _concreteTypeDeserializer = concreteTypeDeserializer;
+            _consumingStrategy += ConsumeViaConcreteDeserializer;
+            _logger.LogDebug($"Initialized consumer from queue: {_sourceQueue.Name}," +
+                $" expecting {typeof(TData).ToString()} derieved data");
+        }
+
+        public void Start()
+        {
+            _consumeTask = _netqBus.Advanced.Consume(_sourceQueue, _consumingStrategy);
+            _logger.LogInformation("Started consumer sucessfully");
+        }
+
+        private void ConsumeViaDeserializer(byte[] messageBytes, MessageProperties messageProperties,
+            MessageReceivedInfo messageInfo)
+        {
+            TData data = _deserializer.Deserialize(messageBytes);
+            RabbitMessageProperties properties = _coverter.Convert(messageProperties);
+            MessageReceived?.Invoke(data);
+            MessageWithMetadataReceived?.Invoke(data, properties);
+        }
+
+        private void ConsumeViaConcreteDeserializer(byte[] messageBytes, MessageProperties messageProperties,
+            MessageReceivedInfo messageInfo)
+        {
+            Type messageConcreteType = messageProperties.Type.GetType();
+            TData data = _concreteTypeDeserializer.Deserialize(messageBytes, messageConcreteType);
+            RabbitMessageProperties properties = _coverter.Convert(messageProperties);
+            MessageReceived?.Invoke(data);
+            MessageWithMetadataReceived?.Invoke(data, properties);
+        }
+
+        public void Stop()
+        {
+            if (_consumeTask == null)
+            {
+                _logger.LogInformation("Couldn't stop consumer, please make sure it was started");
```

this an exception message not an info log message
I think we need to throw an InvalidOperationException with this message or write an error or warnning log for this

---

This may not be an error, but i certainly agree that this is an excellent example of unexpected behavior. I will change it to `warning` dont you think?

---

f